pandas/core/indexes/base.py

```diff
@@ -2745,6 +2745,7 @@ def intersection(self, other, sort=False):
             except TypeError:
                 pass
             else:
+                result = algos.unique1d(result)
```

Comment:
how are these non-unique at this point? does _inner_indexer not do this?

Reply:
No, for lvals ``[1, 2, 2, 3]`` and rvals ``[3,3]`` we get ``[3, 3]`` not ``[3]``- But we can not change this without problems at other places, because this function is also used in inner joins, where ``[3, 3]`` is expected.